lib/tokenFactory.js

```diff
@@ -0,0 +1,116 @@
+const Token = require('./token');
+const iron = require('iron');
+const nodeify = require('./nodeify');
+// Symbols for private fields
+const password = Symbol('password');
+
+let instance;
+
+/**
+ * Seal token value
+ * @method sealToken
+ * @param  {String}         tokenValue     Token value to seal
+ * @param  {String}         pw             Password to seal with
+ * @return {Promise}
+ */
+function sealToken(tokenValue, pw) {
+    return nodeify.withContext(iron, 'seal', [tokenValue, pw, iron.defaults]);
+}
+
+/**
+ * Unseal token value and return the token with the unsealed value
+ * @param  {TokenModel}         token   Token to unseal
+ * @param  {String}             pw      Password to unseal with[description]
+ * @return {Promise}
+ */
+function unsealToken(token, pw) {
+    return nodeify.withContext(iron, 'unseal', [token.value, pw, iron.defaults])
+        .then((unsealed) => {
+            token.value = unsealed;
+
+            return token;
+        });
+}
+
+class TokenFactory extends BaseFactory {
+    /**
+     * Construct a TokenFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore     Object that will perform operations on the datastore
+     * @param  {String}    config.password      Password for encryption operations
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Instantiate a Token class
+     * @method createClass
+     * @param  config
+     * @return {Token}
+     */
+    createClass(config) {
+        const c = config;
+
+        c.password = this[password];
+
+        return new Token(c);
+    }
+
+    /**
+     * Create a token model
+     * Need to seal the token before saving
+     * @method create
+     * @param  {Object}     config
+     * @param  {String}     config.userId            The ID of the associated user
+     * @param  {String}     config.value             The token value
+     * @param  {String}     config.name              The token name
+     * @param  {String}     config.description       The token description
+     * @return {Promise}
+     */
+    create(config) {
+        return sealToken(config.value, this[password])
+            .then((sealed) => {
+                config.value = sealed;
+                config.lastUsed = null;
+
+                return super.create(config);
+            });
+    }
+
+    /**
+     * List tokens with pagination and filter options
+     * @method list
+     * @param  {Object}   config                  Config object
+     * @param  {Object}   config.params           Parameters to filter on
+     * @param  {Object}   config.paginate         Pagination parameters
+     * @param  {Number}   config.paginate.count   Number of items per page
+     * @param  {Number}   config.paginate.page    Specific page of the set to return
+     * @return {Promise}
+     */
+    list(config) {
```

Comment:
We should never expose the token value to the user except after a create (or regenerate?).